inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -303,79 +279,97 @@ def _create_repeated_schedule(self, rs: RepeatedSchedule) -> None:
                 raise DispatcherException(
                     "No schedule id was added to the repeated_schedule table.")
 
-            logger.debug(f"Added repeated schedule with id: {str(schedule_id)}, request_id:{rs.request_id}, cron_duration: {rs.cron_duration}, cron_minutes: {rs.cron_minutes}, cron_hours: {rs.cron_hours}, cron_day_month: {rs.cron_day_month}, cron_month: {rs.cron_month}, cron_day_week: {rs.cron_day_week}")  # noqa
+            logger.debug(f"Added repeated schedule with id: {str(schedule_id)}, request_id:{rs.request_id}, job_id:{rs.job_id}, cron_duration: {rs.cron_duration}, cron_minutes: {rs.cron_minutes}, cron_hours: {rs.cron_hours}, cron_day_month: {rs.cron_day_month}, cron_month: {rs.cron_month}, cron_day_week: {rs.cron_day_week}")  # noqa
 
-            # Add the jobs to the JOB table
-            job_ids = self._insert_job(rs.manifests)
+            if rs.job_id:
+                # Add the jobs to the JOB table
+                task_ids = self._insert_job(rs.job_id, rs.manifests)
+                # Add the schedule_id and job_id to the repeated_schedule_job table
+                self._insert_repeated_schedule_job_tables(schedule_id, task_ids)
 
-            # Add the schedule_id and job_id to the repeated_schedule_job table
-            self._insert_repeated_schedule_job_tables(schedule_id, job_ids)
             self._conn.commit()
         except (sqlite3.Error) as e:
             self._conn.rollback()
             logger.error(f"Transaction failed: {str(e)}")
             raise DispatcherException(f"Transaction failed: {str(e)}")
 
-    def _insert_job(self, manifests: list[str]) -> list[int]:
+    def _insert_job(self, job_id: str, manifests: list[str]) -> list[int]:
         # Add the job to the job table
         if len(manifests) == 0:
             raise DispatcherException(
                 "Error: At least one job is required for the schedule.  Jobs list is empty.")
 
-        job_ids: list[int] = []
+        task_ids: list[int] = []
 
         for manifest in manifests:
-            sql = ''' INSERT INTO job(manifest) VALUES(?); '''
+            sql = ''' INSERT INTO job(job_id, manifest) VALUES(?,?); '''
```

Comment:
<h3>$\color{#15F4EE}Miscellaneous$</h3>

In the method `_insert_job`, ensure that `job_id` is unique before insertion to avoid potential conflicts.

Reply:
It won't be unique since it's being added with multiple manifests.  The auto-incremented task-id ensure the tuples are unique.